vassal-app/src/main/java/VASSAL/build/GameModule.java

```diff
@@ -337,18 +341,29 @@ protected void build() throws IOException {
       build(null);
     }
     else {
+      final AbstractMetaData data = MetaDataFactory.buildMetaData(f);
+      final String properBuildFileName = (VersionUtils.compareVersions(data.getVassalVersion(), "3.4.999") < 0) ? BUILDFILE_OLD : BUILDFILE;
```

Comment:
Is there a reason you can't do `VersionUtils.compareVersions(data.getVassalVersion(), "3.5")`? That ought to be negative for anything 3.4 or earlier.

Reply:
That's actually what I had in the original version, I think, and then 3.5.0-SNAPSHOT turned out to be earlier?

Reply:
I need a version that's either "guaranteed lower than ANY OTHER 3.5" or "guaranteed higher than ANY OTHER 3.4".

Reply:
Ah, crud, right. Then you want to use `VersionUtils.trunctateToMinorVersion()` on `data.getVassalVersion()` first, to cut off anything past the minor version.